device/iot-device-samples/device-connection-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceClientManager.java

```diff
@@ -0,0 +1,415 @@
+        }
+    }
+
+    /**
+     * Sends telemetry and updates reported properties periodically. Returns if the client's connection was lost during this.
+     *
+     * This method simulates typical useage of a client, but can be modified to fit your use case without losing the
+     * automatic reconnection and retry that this sample has.
+     *
+     * @throws IllegalStateException If the client's connection was lost as this method attempted to send telemetry or
+     * update reported properties.
+     */
+    private void doWork() throws IllegalStateException
+    {
+        while (!reconnectNeeded)
+        {
+            sendTelemetryAsync();
+            updateReportedPropertiesAsync();
+
+            try
+            {
+                // not recommended for actual applications, but this makes the sample's logs print at a readable pace
+                Thread.sleep(10000);
+            }
+            catch (InterruptedException e)
+            {
+                System.out.println("Worker thread interrupted, likely because the sample is being stopped.");
+                return;
+            }
+        }
+    }
+
+    // region telemetry
+    private void sendTelemetryAsync()
+    {
+        Message messageToSend;
+        if (this.telemetryToResend.isEmpty())
+        {
+            // if no previous messages failed to send, send a new message
+            messageToSend = new Message("hello world");
+        }
+        else
+        {
+            // if any previous message failed to send, retry sending it before moving on to new messages
+            messageToSend = this.telemetryToResend.remove();
+        }
+
+        try
+        {
+            this.deviceClient.sendEventAsync(messageToSend, this, messageToSend);
+        }
+        catch (IllegalStateException e)
+        {
+            System.out.println("Device client was closed, so requeueing the message locally");
+            this.telemetryToResend.add(messageToSend);
+            throw e;
+        }
+    }
+
+    // callback for when a telemetry message is sent
+    @Override
+    public void execute(IotHubStatusCode responseStatus, IotHubClientException e, Object callbackContext)
+    {
+        Message message = (Message) callbackContext;
+        if (e == null)
+        {
+            System.out.println("Successfully sent message with correlation id " + message.getCorrelationId());
+        }
+        else if (e.isRetryable())
+        {
+            System.out.println("Failed to send message with correlation id " + message.getCorrelationId() + " due to retryable error with status code " + responseStatus.name() + ". Requeueing message.");
+            telemetryToResend.add(message);
+        }
+        else
+        {
+            System.out.println("Failed to send message with correlation id " + message.getCorrelationId() + " due to an unretryable error with status code " + responseStatus.name() + ". Discarding message as it can never be sent");
```

Comment:
```suggestion
            System.out.println("Failed to send message with correlation Id " + message.getCorrelationId() + " due to an unretryable error with status code " + responseStatus.name() + ". Discarding message as it can never be sent");
```